client/scripts/controllers/app/recent_activity.ts

```diff
@@ -0,0 +1,57 @@
+import { OffchainThread, AddressInfo, OffchainComment, OffchainReaction } from 'models';
```

Comment:
How are we keeping the OffchainThread objects stored here in sync with those stored on the threads controller?Are we doing so at all? We should explain in a comment at the top of this controller.

Reply:
I'm not quite sure what you mean. They're modeledFromServer, so their format is "in sync." When a thread is deleted or created in the threads controller, it also updates the recentActivity controller. But the actual state objects contain very different sets of data... And threads controller only grabs threads when you visit a community, so stuff like splashpage access is totally missing.

Reply:
Okay, that seems all right!